skimage/draw/draw.py

```diff
@@ -520,7 +554,20 @@ def circle_perimeter(r, c, radius, method='bresenham', shape=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
     """
-    return _circle_perimeter(r, c, radius, method, shape)
+    if not np.can_cast(r, np.integer) or not np.can_cast(c, np.integer):
+        raise TypeError("r and c must be integers")
+
+    rr, cc = _circle_perimeter(radius, method)
+    rr, cc = np.unique(np.stack([rr, cc]), axis=1)
```

Comment:
Is this the key line in all of this?

Reply:
Yes, this as well as the other two similar lines are the ones addressing the duplicates returned by the Cython implementations. The rest is just maintenance that's not strictly necessary but, as I felt, simplifies the Cython side a little.

Reply:
Got it. That makes sense. I just wanted to make sure I didn't miss anything else